addons/udes_stock/models/res_users.py

```diff
@@ -37,3 +67,59 @@ def get_user_warehouse(self, aux_domain=None):
                 )
             )
         return warehouse
+
+    def assign_picking_to_users(self, picking):
+        """
+        Assign a picking to user(s) in self
+        """
+        picking.ensure_one()
+        picking_type = picking.picking_type_id
+
+        # Find the users to assign
+        users_to_assign = self.filtered(lambda u: u.u_picking_id != picking)
+        if users_to_assign:
+            if picking.state in NON_ASSIGNABLE_STATES:
+                _logger.warning(
+                    "Cannot assign users %s to picking %s as it is in state %s"
+                    % (self.mapped("name"), picking.name, picking.state)
+                )
+            elif len(self) > 1 and not picking_type.can_handle_multiple_users():
+                _logger.warning(
+                    "Cannot assign users %s to picking %s as picking type %s only allows one user"
+                    % (self.mapped("name"), picking.name, picking_type.name)
+                )
+            else:
+                # Un assign the current users from their current (different) picking
+                users_to_assign.un_assign_pickings_from_users(new_picking=picking)
+                # Kick out any users from the current picking if needed
+                # Do not kick the current users in self out as they are already unassigned
+                # from their own pickings.
+                # NOTE: We could check here if there are any users in the picking, and kick them
+                # out if needed, but instead we just extend unassign_users, as this
+                # is needed when closing batches or if we extend the behaviour of multiple users
+                # working on a picking.
+                picking.un_assign_users(skip_users=users_to_assign)
+                users_to_assign._assign_picking_to_users(picking)
+
+    def _assign_picking_to_users(self, picking):
+        """Placing into a method in order to be able to inherit and change behaviour if needed.
+        If the picking has never been started, update the start time."""
+        now = PreciseDatetime.now()
+        self.sudo().write(
+            {"u_picking_id": picking.id, "u_picking_assigned_precise_time": now})
+        if not picking.u_date_started:
+            picking.write({"u_date_started": now})
+
+    def un_assign_pickings_from_users(self, new_picking=False):
```

Comment:
"unassign" is better here I think; in written English the "un" and "assign" are not separated.

Reply:
Not very confident with English, trustet Pycharm (it is showing underlined which shows when it's not in English) and googled after and it shows that should be separated, thats why changed the name

Reply:
Some of the tests still have `un_assign` in their names

Reply:
Checked all strings where was un_assign, i think there isnt any left